tools/devtool

```diff
@@ -375,6 +388,102 @@ run_devctr() {
         "$DEVCTR_IMAGE" "${ctr_args[@]}"
 }
 
+# Given a list of packages it adds them to {arch-based}-requirements_freeze.txt
+#
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+#
+perform_update() {
```

Comment:
All of these helper functions should include pip/python in their names. Their logic is specific to pip, not to all packages installed in the docker container

Reply:
these functions also have very similar, generic names. It's hard to tell the difference from `update_packages` and `perform_update` without reading the code. You can try being a little more specific, where it makes sense

Reply:
fixed